hugr-llvm/src/extension/collections/static_array.rs

```diff
@@ -0,0 +1,517 @@
+                {
+                    let sac = self.0.clone();
+                    move |ts, custom_type| {
+                        let element_type = custom_type.args()[0]
+                            .as_type()
+                            .expect("Type argument for static array must be a type");
+                        sac.static_array_type(ts, &element_type)
+                    }
+                },
+            )
+            .custom_const::<StaticArrayValue>({
+                let sac = self.0.clone();
+                move |context, sav| sac.static_array_value(context, sav)
+            })
+            .simple_extension_op::<StaticArrayOpDef>({
+                let sac = self.0.clone();
+                move |context, args, op| {
+                    let op = op.instantiate(args.node().args())?;
+                    sac.static_array_op(context, args, op)
+                }
+            })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use float_types::float64_type;
+    use hugr_core::extension::prelude::ConstUsize;
+    use hugr_core::ops::constant::CustomConst;
+    use hugr_core::ops::OpType;
+    use hugr_core::ops::Value;
+    use hugr_core::std_extensions::arithmetic::float_types::{self, ConstF64};
+    use rstest::rstest;
+
+    use hugr_core::extension::simple_op::MakeRegisteredOp;
+    use hugr_core::extension::{prelude::bool_t, ExtensionRegistry};
+
+    use crate::check_emission;
+    use crate::test::single_op_hugr;
+    use crate::{
+        emit::test::SimpleHugrConfig,
+        test::{exec_ctx, llvm_ctx, TestContext},
+    };
+    use hugr_core::builder::{Dataflow as _, DataflowSubContainer as _};
+
+    #[rstest]
+    #[case(0, StaticArrayOpDef::get, usize_t())]
+    #[case(1, StaticArrayOpDef::get, bool_t())]
+    #[case(2, StaticArrayOpDef::len, usize_t())]
+    #[case(3, StaticArrayOpDef::len, bool_t())]
+    fn static_array_op_codegen(
+        #[case] _i: i32,
+        #[with(_i)] mut llvm_ctx: TestContext,
+        #[case] op: StaticArrayOpDef,
+        #[case] ty: HugrType,
+    ) {
+        let op = op.instantiate(&[ty.clone().into()]).unwrap();
+        let op = OpType::from(op.to_extension_op().unwrap());
+        llvm_ctx.add_extensions(|ceb| {
+            ceb.add_default_static_array_extensions()
+                .add_default_prelude_extensions()
+        });
+        let hugr = single_op_hugr(op);
+        check_emission!(hugr, llvm_ctx);
+    }
+
+    #[rstest]
+    #[case(0, StaticArrayValue::try_new("a", usize_t(), (0..10).map(|x| ConstUsize::new(x).into())).unwrap())]
+    #[case(1, StaticArrayValue::try_new("b", float64_type(), (0..10).map(|x| ConstF64::new(x as f64).into())).unwrap())]
+    #[case(2, StaticArrayValue::try_new("c", bool_t(), (0..10).map(|x| Value::from_bool(x % 2 == 0))).unwrap())]
+    #[case(3, StaticArrayValue::try_new("d", option_type(usize_t()).into(), (0..10).map(|x| Value::some([ConstUsize::new(x)]))).unwrap())]
+    fn static_array_const_codegen(
```

Comment:
This and the following test fail when I run
```
cargo test --all-features --workspace
```
with the error
```
thread 'extension::collections::static_array::test::static_array_const_codegen::case_1' panicked at hugr-llvm/src/emit/test.rs:177:48:
Parent node Node(1) has extensions [arithmetic.conversions, arithmetic.float, arithmetic.float.types, arithmetic.int, arithmetic.int.types, collections.array, collections.list, logic, prelude] that are too restrictive for child node Node(4), they must include child extensions [collections.static_array, prelude]
```

I guess we need to annotate them so that they don't run with the `extension_inference` feature?